internal/gitprovider/azure/azure.go

```diff
@@ -0,0 +1,234 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"slices"
+	"strings"
+
+	"github.com/akuity/kargo/internal/git"
+	"github.com/akuity/kargo/internal/gitprovider"
+	"k8s.io/utils/ptr"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7"
+	adogit "github.com/microsoft/azure-devops-go-api/azuredevops/v7/git"
+)
+
+const ProviderName = "azure"
+
+// Azure DevOps URLs can be of two different forms:
+//
+//   - https://dev.azure.com/org/project/_git/repo
+//   - https://org.visualstudio.com/project/_git/repo
+//
+// We support both forms.
+var providerSuffixes = []string{"dev.azure.com", "visualstudio.com"}
```

Comment:
Nit: I spent some time wondering why the example hostname was `org.visualstudio.com`, but the suffix was `visualstudio.com`... then realized "org" was a placeholder. I think it would make what's happening here just slightly more clear if you did this:

```suggestion
// Azure DevOps URLs can be of two different forms:
//
//   - https://dev.azure.com/org/<project>/_git/<repo>
//   - https://<org>.visualstudio.com/<project>/_git/<repo>
//
// We support both forms.
var providerSuffixes = []string{"dev.azure.com", "visualstudio.com"}
```

Reply:
done